python/python/lance/dataset.py

```diff
@@ -1570,37 +1583,32 @@ def with_fragments(
     def nearest(
         self,
         column: str,
-        q: pa.FloatingPointArray | List[float] | np.ndarray,
+        q: QueryVectorLike,
         k: Optional[int] = None,
         metric: Optional[str] = None,
         nprobes: Optional[int] = None,
         refine_factor: Optional[int] = None,
         use_index: bool = True,
     ) -> ScannerBuilder:
-        column_field = self.ds.schema.field(column)
-        q_size = q.size if isinstance(q, np.ndarray) else len(q)
+        q = _coerce_query_vector(q)
 
         if self.ds.schema.get_field_index(column) < 0:
-            raise ValueError(f"Embedding column {column} not in dataset")
-        if not (
-            isinstance(
-                column_field.type, (pa.FloatingPointArray, np.ndarray, list, tuple)
-            )
-            or pa.types.is_fixed_size_list(column_field.type)
-        ):
+            raise ValueError(f"Embedding column {column} is not in the dataset")
+
+        column_field = self.ds.schema.field(column)
+        column_type = column_field.type
+        if hasattr(column_type, "storage_type"):
```

Comment:
Do we still need to check storage type for pyarrow > 12?

Reply:
Not for fixed shape tensor array.  Once we are using pyarrow 13 we can directly check for `FixedShapeTensorArray`.  However, another advantage of using `storage_type` is that we are future proofing ourselves against other possible extension types.